xapian-core/include/xapian/cluster.h

```diff
@@ -316,10 +407,30 @@ class XAPIAN_VISIBILITY_DEFAULT RoundRobin {
     ~RoundRobin();
 
     // Implements RoundRobin clustering
-    ClusterSet cluster(MSet &mset, unsigned int k);
+    ClusterSet cluster(MSet &mset, unsigned int num_of_clusters);
 
     // Returns the description of the clusterer being used
     std::string get_description();
 };
-}
+
+/** Kmeans clusterer:
+ *  This clusterer implements the K-Means clustering algorithm
+ */
+class XAPIAN_VISIBILITY_DEFAULT KMeans {
+    std::vector<Point> docs;
+    //std::map<int, docid> doc_list;
+    std::vector<Point> centroids;
+  public:
+    void initialize_centroids(const unsigned int k, ClusterSet &cset);
+
+    void initialize_points(MSetDocumentSource docs, TermListGroup tlg);
+
+    // Implements KMeans clustering
+    ClusterSet cluster(MSet mset, unsigned int k);
```

Comment:
If it's a _property_ of a `Clusterer` then I suggest you pass it in the constructor, not to the `cluster()` method. There are clustering algorithms that don't need to know in advance how many clusters they're generating, so that's probably a better approach.